packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -174,34 +174,23 @@ async function _getDashboardStats(app) {
     }
   }
 
-  const poolIds = new Set()
-  const hosts = []
-  const writableSrs = []
-  const nonReplicaVms = []
+  const pools = Object.values(app.objects.indexes.type.pool)
+  const hosts = Object.values(app.objects.indexes.type.host)
+  const srs = Object.values(app.objects.indexes.type.SR)
+  const vms = Object.values(app.objects.indexes.type.VM)
```

Comment:
Be careful, any one of these entries can be `undefined` if there are no such objects of a given type.